File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/DomainConverter.java
package com.platdmit.simplecloudmanager.domain.converters;

import com.platdmit.simplecloudmanager.data.api.models.ApiDomain;
import com.platdmit.simplecloudmanager.data.database.entity.DbDomain;
import com.platdmit.simplecloudmanager.domain.models.Domain;

import java.util.List;

public interface DomainConverter {
    DbDomain fromApiToDb(ApiDomain apiDomain);
    Domain fromDbToDomain(DbDomain dbDomain);
    Domain fromApiToDomain(ApiDomain apiDomain);
    List<Domain> fromApiToDomainList(List<ApiDomain> apiList);
    List<Domain> fromDbToDomainList(List<DbDomain> dbList);
    List<DbDomain> fromApiToDbList(List<ApiDomain> apiList);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/DomainConverter.kt
package com.platdmit.simplecloudmanager.domain.converters

import com.platdmit.data.api.models.ApiDomain
import com.platdmit.data.database.entity.DbDomain
import com.platdmit.simplecloudmanager.domain.models.Domain

interface DomainConverter {
    fun fromApiToDb(apiDomain: ApiDomain): DbDomain
    fun fromDbToDomain(dbDomain: DbDomain): Domain
    fun fromApiToDomain(apiDomain: ApiDomain): Domain
    fun fromApiToDomainList(apiList: List<ApiDomain>): List<Domain>
    fun fromDbToDomainList(dbList: List<DbDomain>): List<Domain>
    fun fromApiToDbList(apiList: List<ApiDomain>): List<DbDomain>
}